Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -173,180 +153,207 @@ public static class CustomEntitySearch
+                                dynamicDistanceCalc[currpotentialEntityMatchIndex, currWordIndex + 1] + accentCost, // deletion
+                                dynamicDistanceCalc[currpotentialEntityMatchIndex + 1, currWordIndex] + accentCost), // insertion
+                                dynamicDistanceCalc[currpotentialEntityMatchIndex, currWordIndex] + cost); // substitution
+                   
+                    if (currpotentialEntityMatchIndex > 0 && currWordIndex > 0 && potentialMatch[currpotentialEntityMatchIndex].Equals(entityToFind[currWordIndex - 1]) && entityToFind[currWordIndex].Equals(potentialMatch[currpotentialEntityMatchIndex - 1]))
+                    {
+                        dynamicDistanceCalc[currpotentialEntityMatchIndex + 1, currWordIndex + 1] = 
+                            Math.Min(
+                                dynamicDistanceCalc[currpotentialEntityMatchIndex + 1, currWordIndex + 1],
+                                dynamicDistanceCalc[currpotentialEntityMatchIndex - 1, currWordIndex - 1] + cost); // transposition
+                    }
                 }
             }
 
-            return dynamicDistanceCalc[text.Length, checkMatch.Length];
+            return dynamicDistanceCalc[potentialMatch.Length, entityToFind.Length];
         }
 
-        /*
-        * Given an entity the user wants to find, this method removes delineating characters if they are found in the
-        * beginning or end of the entity definition. The method then returns the exact word that will be used for fuzzy matching
-        */
-        public static string CreateWordArray(string checkMatch)
+        /// <summary>
+        /// Trims delineating characters from the beginning and end of some text
+        /// </summary>
+        /// <param name="textToTrim">the text to trim</param>
+        /// <returns>a string with the delineating characters trimmed from the beginning and end</returns>
+        public static string TrimDelineatingCharacters(string textToTrim)
         {
-            int initCheckIndex = 0;
-            int endCheckIndex = checkMatch.Length - 1;
-
-            while (initCheckIndex < checkMatch.Length && checkMatch[initCheckIndex].IsDelineating())
-                initCheckIndex++;
-            while (endCheckIndex >= 0 && checkMatch[endCheckIndex].IsDelineating())
-                endCheckIndex--;
-            if (initCheckIndex != 0 || endCheckIndex != checkMatch.Length - 1)
+            int startIndex = 0;
+            int endIndex = textToTrim.Length - 1;
+
+            while (startIndex < textToTrim.Length && textToTrim[startIndex].IsDelineating())
+                startIndex++;
+            while (endIndex >= 0 && textToTrim[endIndex].IsDelineating())
+                endIndex--;
+
+            if (startIndex >= endIndex)
             {
-                return checkMatch.Substring(initCheckIndex, endCheckIndex - initCheckIndex + 1);
+                return string.Empty;
```

Comment:
nit: `""`

Reply:
eh, i prefer string.Empty. Makes it explicit that i intended it to be empty. Let me know if you feel strongly about "" though

---
In reply to: [330769371](https://github.com/Azure-Samples/azure-search-power-skills/pull/14#discussion_r330769371) [](ancestors = 330769371)

Reply:
I mildly feel `""` is always better, and a little more strongly we should be consistent throughout this codebase, but eh, such a small and inconsequential detail...